src/partition/src/error.rs

```diff
@@ -136,6 +150,9 @@ impl ErrorExt for Error {
             Error::InvalidTableRouteData { .. } => StatusCode::Internal,
             Error::ConvertScalarValue { .. } => StatusCode::Internal,
             Error::FindDatanode { .. } => StatusCode::InvalidArguments,
+            Error::CreateDefaultToRead { .. } | Error::MissingDefaultValue { .. } => {
```

Comment:
I think we can use `CreateDefaultToRead`'s `source`'s status, like this: `Error::CreateDefaultToRead { source, .. } => source.status_code()`